### PR TITLE
update github sponsors tax document link description

### LINKED ISSUE #33802: GitHub Sponsors tax document link description is incorrect
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/.github/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://github.com/github/docs/blob/main/content/sponsors/receiving-sponsorships-through-github-sponsors/setting-up-github-sponsors-for-your-personal-account.md, via https://github.com/github/docs/blob/main/data/reusables/sponsors/tax-form-information-dev.md

### What part(s) of the article would you like to see updated?

The tax form information indicates the the `W-8BEN` is for non-US residents, which is incorrect. It is for non-US residents *and citizens*, whereas `W-9` is for US residents and non-resident US citizens (it may also apply to non-US citizens with resident status even if they are currently non-resident, but that is an odd situation).

US expats must use `W-9`, not `W-8BEN`.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
### Why:

Update github sponsors tax document link description

Closes: #33802 

### Check off the following:

- [ ] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [ ] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@gagan-bhullar-tech Hello! 👋 Thank you for opening a PR!

We will need to put this PR on hold for now until the issue this PR is linked to has been reviewed by the team 💛

**Comment by ardamcan** [comment]
👌🥇💣

File: data/reusables/sponsors/tax-form-information-dev.md
Comment (catsintents) [inline data/reusables/sponsors/tax-form-information-dev.md:3]: This line is correct.

File: data/reusables/sponsors/tax-form-information-dev.md
Comment (catsintents) [inline data/reusables/sponsors/tax-form-information-dev.md:4]: I am not sure we need to merge this one.

**Review by catsintents — CHANGES_REQUESTED** [review]

File: data/reusables/sponsors/tax-form-information-dev.md
Comment (slincoln75) [reply data/reusables/sponsors/tax-form-information-dev.md:3]: This looks good to me.

File: data/reusables/sponsors/tax-form-information-dev.md
Comment (nguyenalex836) [reply data/reusables/sponsors/tax-form-information-dev.md:4]: @catsintents @slincoln75 Thank you both for these comments! 

@gagan-bhullar-tech Would you be able to update the PR per the comments above? Once that's complete, we merge this PR 💛

Comment (gagan-bhullar-tech) [reply data/reusables/sponsors/tax-form-information-dev.md]: @nguyenalex836 its updated.

**Review by nguyenalex836 — APPROVED** [review]

**Pull request merged**
